pkgs/development/python-modules/django-markdownx/default.nix

```diff
@@ -0,0 +1,50 @@
+{ lib
+, buildPythonPackage
+, django
+, fetchFromGitHub
+, markdown
+, pillow
+, python
+, pythonOlder
+}:
+
+buildPythonPackage rec {
+  pname = "django-markdownx";
+  version = "4.0.7";
+  format = "setuptools";
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "neutronX";
+    repo = "django-markdownx";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-FZPUlogVd3FMGeH1vfKHA3tXVps0ET+UCQJflpiV2lE=";
+  };
+
+  postPatch = ''
+    substituteInPlace setup.py \
+      --replace "install_requires=get_requirements()," ""
+  '';
+
+  propagatedBuildInputs = [
+    django
+    markdown
+    pillow
+  ];
+
+  # tests only executeable in vagrant
```

Comment:
There are some examples of tests for django libraries in nixpkgs, but in fact, it is a bit tricky.
This may be more complicated.😢

> Django packages require `manage.py` and more often than not, `settings.py` files to run. This introduces a challenge for testing apps outside of a fully constructed project environment.

https://github.com/neutronX/django-markdownx/blob/1ced4688efab60f22cdff323e8010e2af8108960/docs-src/contributions.md?plain=1#L68

Reply:
this is probably not going to work out, since the tests require docker or vegrant